src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java

```diff
@@ -0,0 +1,248 @@
+                        + "\", type=\"activity\"")
+                .build();
+      }
+      poller =
+          new Poller<>(
+              options.getIdentity(),
+              laPollTask,
+              new PollTaskExecutor<>(domain, taskList, options, new TaskHandlerImpl(handler)),
+              pollerOptions,
+              options.getMetricsScope());
+      poller.start();
+      options.getMetricsScope().counter(MetricsType.WORKER_START_COUNTER).inc(1);
+    }
+  }
+
+  @Override
+  public boolean isStarted() {
+    return poller.isStarted();
+  }
+
+  @Override
+  public boolean isShutdown() {
+    return poller.isShutdown();
+  }
+
+  @Override
+  public boolean isTerminated() {
+    return poller.isTerminated();
+  }
+
+  @Override
+  public void shutdown() {
+    poller.shutdown();
+  }
+
+  @Override
+  public void shutdownNow() {
+    poller.shutdownNow();
+  }
+
+  @Override
+  public void awaitTermination(long timeout, TimeUnit unit) {
+    poller.awaitTermination(timeout, unit);
+  }
+
+  @Override
+  public void suspendPolling() {
+    poller.suspendPolling();
+  }
+
+  @Override
+  public void resumePolling() {
+    poller.resumePolling();
+  }
+
+  @Override
+  public boolean isSuspended() {
+    return poller.isSuspended();
+  }
+
+  public static class Task {
+    private final ExecuteLocalActivityParameters params;
+    private final Consumer<HistoryEvent> eventConsumer;
+    private final LongSupplier currentTimeMillis;
+    private final LongSupplier replayTimeUpdatedAtMillis;
+    long taskStartTime;
+    private final int decisionTimeoutSeconds;
+
+    public Task(
+        ExecuteLocalActivityParameters params,
+        Consumer<HistoryEvent> eventConsumer,
+        int decisionTimeoutSeconds,
+        LongSupplier currentTimeMillis,
+        LongSupplier replayTimeUpdatedAtMillis) {
+      this.params = params;
+      this.eventConsumer = eventConsumer;
+      this.currentTimeMillis = currentTimeMillis;
+      this.replayTimeUpdatedAtMillis = replayTimeUpdatedAtMillis;
+      this.decisionTimeoutSeconds = decisionTimeoutSeconds;
+    }
+  }
+
+  public Consumer<LocalActivityWorker.Task> getLocalActivityTaskConsumer() {
+    return laPollTask;
+  }
+
+  private class TaskHandlerImpl implements PollTaskExecutor.TaskHandler<Task> {
+
+    final ActivityTaskHandler handler;
+
+    private TaskHandlerImpl(ActivityTaskHandler handler) {
+      this.handler = handler;
+    }
+
+    @Override
+    public void handle(Task task) throws Exception {
+      task.taskStartTime = System.currentTimeMillis();
+      ActivityTaskHandler.Result result = handleLocalActivity(task);
+
+      ClockDecisionContext.LocalActivityMarkerData marker;
+      long replayTimeMillis =
+          task.currentTimeMillis.getAsLong()
+              + (System.currentTimeMillis() - task.replayTimeUpdatedAtMillis.getAsLong());
+      if (result.getTaskCompleted() != null) {
+        marker =
+            new ClockDecisionContext.LocalActivityMarkerData(
```

Comment:
May be use Builder pattern here? The number of args is too large.

Reply:
Done